src/components/maps.js

```diff
@@ -0,0 +1,120 @@
+// src/pages/MapPage.jsx
+import React, { useEffect, useState } from 'react'
+import { useParams, useNavigate } from 'react-router-dom'
+import { MapContainer, TileLayer, Marker, Popup, useMap } from 'react-leaflet'
+import redMarker from "../images/marker/red_marker.png"
+import burger from "../images/category/burger.png"
+import pizza from "../images/category/pizza.png"
+import fried_chicken from "../images/category/fried_chicken.png"
+import steak from "../images/category/steak.png"
+import desserts from "../images/category/desserts.png"
+import { Icon } from "leaflet"
+import axios from 'axios'
+import 'leaflet/dist/leaflet.css'
+
+const MapPage = () => {
+    const { state } = useParams()
+    const [restaurants, setRestaurants] = useState([])
+    const navigate = useNavigate
+
+    const customIcon = new Icon({
+        iconUrl: redMarker,
+        iconSize: [40, 40]
+    })
+
+    useEffect(() => {
+        const fetchMenuData = async () => {
+            try {
+                const response = await axios.get('https://menus-api.vercel.app/')
+                const data = response.data
+                const allRestaurantData = Object.values(data)
+        
+                // Normalize all data is Array
+                const normalized = allRestaurantData.map(item => 
+                    Array.isArray(item) ? item : [item])
+
+                // push data we needed to use in allRestaurants Array
+                const allRestaurants = []
+                normalized.forEach(category => {
+                    category.forEach(restaurant => {
+                      if (restaurant.country) {
+                        const restaurantState = restaurant.country.split(', ').pop()
+                        // no unique id in the data
+                        const id = restaurant.index
```

Comment:
Suggestion - use Math.random() to generate a unique id or uuid library. with index id comes as undefined when I log the restaurants list:

![Screenshot 2024-08-07 at 10 03 24 PM](https://github.com/user-attachments/assets/1054c726-2a73-4a33-9346-ab38ab818467)

Reply:
Thank you. This is such a great idea! I will go ahead do that